### PR TITLE
feature: implementation of Verify Message method

### PR DESCRIPTION
**Pull request by CryptologySAC**
The implementation for the Verify message method.

### CODE REVIEW

File: lib/commands/message/verify.js [hunk 1]
```diff
@@ -1,7 +1,80 @@
...
-const chalk = require('chalk')
+const Joi = require('joi')
+const arkjs = require('arkjs')
+const crypto = require('crypto')
+const network = require('../../services/network')
+const output = require('../../utils/output')
+const networks = require('../../config/networks')
+const schema = {
+  msg: Joi.string().required(),
+  publickey: Joi.string().required(), // TODO hex?
```
Comment (faustbrian) [inline lib/commands/message/verify.js]: it should be `publicKey` and `hex` for validation

File: lib/commands/message/verify.js [hunk 2]
```diff
@@ -1,7 +1,80 @@
...
+    }
+  }
+
+  try {
+    output.setFormat(format)
+
+    // Validate input
+    Joi.validate({
+      msg,
+      publickey,
```
Comment (faustbrian) [inline lib/commands/message/verify.js]: `publicKey`

File: lib/commands/message/verify.js [hunk 3]
```diff
@@ -1,7 +1,80 @@
...
+    }
+    await network.setNetwork(net)
+    arkjs.crypto.setNetworkVersion(network.network.version)
+
+    // Verify the Message
+    let hash = crypto.createHash('sha256');
+    hash = hash.update(Buffer.from(msg, 'utf-8')).digest()
+
+    signature = Buffer.from(signature, 'hex');
+    publickey = Buffer.from(publickey, 'hex');
```
Comment (faustbrian) [inline lib/commands/message/verify.js]: `publicKey`

File: lib/commands/message/verify.js [hunk 4]
```diff
@@ -1,7 +1,80 @@
...
+    const ecpair = arkjs.ECPair.fromPublicKeyBuffer(publickey);
```
Comment (faustbrian) [inline lib/commands/message/verify.js]: `publicKey`

File: lib/commands/message/verify.js [hunk 5]
```diff
@@ -1,7 +1,80 @@
...
+}
+
+/**
+ * @dev Verify <msg> <publickey>.
+ * @param {string} msg To verify
+ * @param {string} publickey Key to verify with
+ * @param {string} signature Signature to verify
+ * @param {object} cmd A JSON object containing the options for this query (network, format, verbose).
+ */
+module.exports = async (msg, publickey, signature, cmd) => {
```
Comment (faustbrian) [inline lib/commands/message/verify.js]: `publicKey`

File: bin/cli [hunk 6]
```diff
@@ -90,21 +90,18 @@ commander
...
-  .command('verify <msg> <publickey>')
-  .description('Verify message <msg> with <publickey>.')
+  .command('verify <msg> <publickey> <signature>')
+  .description('Verify message <msg> with <signature> which was signed with <publickey>.')
   .option('-n, --network <network>', 'Connect to network: [mainnet|devnet]', 'mainnet')
-  .option('-c, --node <node>', 'Connect directly to node <node>.')  
   .option('-f, --format <format>', 'Specify how to format the output [json|table]', 'json')
   .option('-v, --verbose', 'Show verbose logging.')
-  .action(async (msg, publickey, cmd) => message.verify(msg, publickey, cmd))
+  .action(async (msg, publickey, signature, cmd) => message.verify(msg, publickey, signature, cmd))
```
Comment (faustbrian) [inline bin/cli]: `publicKey`

File: lib/commands/message/verify.js (hunk 1 above)
Comment (CryptologySAC) [reply lib/commands/message/verify.js]: thx, I'll update that

**Comment by CryptologySAC** [comment]
Updated, thx

**Pull request merged**